Tools/cases_generator/generate_cases.py

```diff
@@ -48,6 +48,34 @@
 )
 
 
+def effect_size(effect: StackEffect) -> tuple[int, str]:
+    if effect.size:
+        return 0, effect.size
+    else:
+        return 1, ""
+
+
+def list_effect_size(effects: list[StackEffect]) -> tuple[int, str]:
+    numeric = 0
+    symbolic: list[str] = []
+    for effect in effects:
+        diff, sym = effect_size(effect)
+        numeric += diff
+        if sym:
+            symbolic.append(sym)
+    return numeric, " + ".join(symbolic)
```

Comment:
Minor:

It doesn't matter now, but we might want to parenthesize each symbolic term. I could see us maybe doing bitwise operations or something as part of a dimension and being bitten by the operator precedence here.